Allow calc() with combined percentages and lengths for line-height

darinadler opened the pull request:
#### 2946bb7fba02b40105f29744a1600ec192b59a78
<pre>
Allow calc() with combined percentages and lengths for line-height
<a href="https://bugs.webkit.org/show_bug.cgi?id=247007">https://bugs.webkit.org/show_bug.cgi?id=247007</a>
rdar://problem/101546260

Reviewed by Tim Nguyen and Sam Weinig.

* LayoutTests/css3/calc/line-height-expected.txt: Expect PASS.

* LayoutTests/fast/css/calc-with-percent-and-number-in-line-height-crash-expected.txt: Added.
* LayoutTests/fast/css/calc-with-percent-and-number-in-line-height-crash.html: Added.
Adopted this test from Chromium.

* LayoutTests/fast/css/line-height-1-expected.html: Added.
* LayoutTests/fast/css/line-height-1.html: Added.
* LayoutTests/fast/css/line-height-2-expected.html: Added.
* LayoutTests/fast/css/line-height-2.html: Added.
Adopted these tests from Mozilla.

* LayoutTests/fast/css/line-height-basics-expected.txt: Added.
* LayoutTests/fast/css/line-height-basics.html: Added.
Adopted this test from Chromium.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-computed-expected.txt:
Expected many more PASS.

* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertLineHeight): Allow percentage combined with length.

Canonical link: <a href="https://commits.webkit.org/256095@main">https://commits.webkit.org/256095@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09a42ecfed98d266e66cbd26f6f0718866d2a15a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94455 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3633 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27364 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104109 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164384 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98453 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3684 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31841 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86783 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100097 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100126 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2644 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80846 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29674 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84566 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/84146 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72568 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38231 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17970 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36098 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19243 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4208 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39993 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/41878 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41945 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38472 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
This looks fine, though please make sure we render these cases correctly/consistently with other browsers before merging.

darinadler committed:
Allow calc() with combined percentages and lengths for line-height
https://bugs.webkit.org/show_bug.cgi?id=247007
rdar://problem/101546260

Reviewed by Tim Nguyen and Sam Weinig.

* LayoutTests/css3/calc/line-height-expected.txt: Expect PASS.

* LayoutTests/fast/css/calc-with-percent-and-number-in-line-height-crash-expected.txt: Added.
* LayoutTests/fast/css/calc-with-percent-and-number-in-line-height-crash.html: Added.
Adopted this test from Chromium.

* LayoutTests/fast/css/line-height-1-expected.html: Added.
* LayoutTests/fast/css/line-height-1.html: Added.
* LayoutTests/fast/css/line-height-2-expected.html: Added.
* LayoutTests/fast/css/line-height-2.html: Added.
Adopted these tests from Mozilla.

* LayoutTests/fast/css/line-height-basics-expected.txt: Added.
* LayoutTests/fast/css/line-height-basics.html: Added.
Adopted this test from Chromium.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-computed-expected.txt:
Expected many more PASS.

* Source/WebCore/style/StyleBuilderConverter.h:
(WebCore::Style::BuilderConverter::convertLineHeight): Allow percentage combined with length.

Canonical link: https://commits.webkit.org/256095@main
